test/extended/image_ecosystem/s2i_perl.go

```diff
@@ -81,6 +82,27 @@ var _ = g.Describe("[image_ecosystem][perl][Slow] hot deploy for openshift perl
 			err = exutil.WaitForDeploymentConfig(oc.KubeClient(), oc.Client(), oc.Namespace(), dcName, 2, oc)
 			o.Expect(err).NotTo(o.HaveOccurred())
 
+			g.By("waiting for a new endpoint")
+			err = e2e.WaitForEndpoint(oc.KubeFramework().ClientSet, oc.Namespace(), dcName)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Ran into an issue where we'd try to hit the endpoint before it was updated, resulting in
+			// request timeouts against the previous pod's ip.  So make sure the endpoint ip has changed before
+			// hitting it.
+			err = wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+				newEndpoint, err := oc.KubeFramework().ClientSet.Core().Endpoints(oc.Namespace()).Get(dcName, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if err != nil {
+					return false, err
```

Comment:
cannot be hit because of prior `o.Expect(err).NotTo(o.HaveOccurred())`

Reply:
thanks, removed the expects (err is handled outside the wait block)